spec/controllers/v0/veteran_onboardings_controller_spec.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe V0::VeteranOnboardingsController, type: :controller do
+  let(:user) { create(:user, :loa3) }
+  let(:veteran_onboarding) do
+    user_verification = create(:user_verification, idme_uuid: user.identity.idme_uuid)
+    create(:veteran_onboarding, user_account: user_verification.user_account)
+  end
+
+  before do
+    Flipper.enable(:veteran_onboarding_beta_flow, user)
+    sign_in_as(user)
+  end
+
+  describe 'GET #show' do
+    it 'returns a success response' do
+      get :show, params: { id: veteran_onboarding.to_param }
+      expect(response).to be_successful
+    end
+  end
+
+  describe 'PATCH #update' do
+    let(:new_attributes) do
+      { display_onboarding_flow: false }
+    end
+
+    it 'updates the requested veteran_onboarding' do
+      patch :update, params: { id: veteran_onboarding.to_param, veteran_onboarding: new_attributes }
+      veteran_onboarding.reload
+      expect(veteran_onboarding.display_onboarding_flow).to eq(false)
```

Comment:
I think it might be better to check the response's data for the `false` display value instead of reloading and checking the model. This could help protect against anything that might transform that response from the controller. What do you think?

Reply:
That is an interesting idea.  I guess there are two schools of thought here, both with plusses and minuses.

Testing the Response from the controller (PROs):
- testing the actual output of the controller, can catch issues where the model state is correct, but the transformation of the data is not correct
- can make the test more robust in the face of changes on the model.  as long as you respect the data contract, we're good

Testing the Response from the controller (CONs):
- makes test more complex
- if data contract changes, will have to update the test

Testing the Model state (PROs):
- simpler
- changes in the data contract will not impact the test (no need to constantly update tests)

Testing the Model state (CONs):
- doesn't catch issues where model state is correct but response data is incorrect
- more tightly coupled with model behavior

I am inclined to leave the test as-is for now, and as we add more complexity to the controller and model, begin to add new tests against the response.  We are currently only able to toggle a single boolean value using this endpoint.

So, let's do both? As things get more complicated, add tests against the response, but for now just check the value on the model.